sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -418,6 +426,19 @@ class JacksonParser(
       }
     }
 
+    // When the input schema is setting to `nullable = false`, make sure the field is not null.
```

Comment:
Does this additional check cause no performance regression? For example, if a schema does not have non-nullable fields at all, we don't need this check, right?

Reply:
Yes, you are right. this check has two purposes.
1. check the non-nullable field. 
2. skipping row using the missing field.

Do you have a better idea to implement that without performance regression?

Reply:
skip this additional check when the mode is `PERMISSIVE`.